internal/fs/inode/dir.go

```diff
@@ -675,13 +675,18 @@ func (d *dirInode) readObjects(
 		// directory "foo/" coexist, the directory would eventually occupy
 		// the value of records["foo"].
 		if strings.HasSuffix(o.Name, "/") {
-			dirName := NewDirName(d.Name(), nameBase)
-			explicitDir := &Core{
-				Bucket:    d.Bucket(),
-				FullName:  dirName,
-				MinObject: storageutil.ConvertObjToMinObject(o),
+			// In a hierarchical bucket, create a folder entry instead of a minObject for each prefix.
+			// This is because in a hierarchical bucket, every directory is considered a folder.
+			// Adding folder entries while looping to through CollapsedRuns instead of here to avoid duplicate entries.
```

Comment:
I understand, but we are already adding the same entries in the CollapsedRuns list. The point is that a directory may or may not be an object in HNS, but it will definitely remain as a prefix/folder. To ensure consistency and avoid duplicate entries, I added the logic below.

Reply:
Got it. Thanks.